org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/linkedediting/LSPLinkedEditingBase.java

```diff
@@ -66,18 +67,10 @@ protected CompletableFuture<LinkedEditingRanges> collectLinkedEditingRanges(IDoc
 		final var identifier = LSPEclipseUtils.toTextDocumentIdentifier(uri);
 		final var params = new TextDocumentPositionParams(identifier, position);
 
-		final var range = new AtomicReference<LinkedEditingRanges>();
-		return LanguageServiceAccessor.getLanguageServers(document,
-					capabilities -> LSPEclipseUtils.hasCapability(capabilities.getLinkedEditingRangeProvider()))
-				.thenComposeAsync(languageServers ->
-					CompletableFuture.allOf(languageServers.stream()
-							.map(ls -> ls.getTextDocumentService().linkedEditingRange(LSPEclipseUtils.toLinkedEditingRangeParams(params))
-								.thenAcceptAsync(result -> {
-									if (result != null && result.getRanges() != null && result.getRanges().size() > 1) {
-										range.set(result);
-									}
-							})).toArray(CompletableFuture[]::new)))
-				.thenApplyAsync(theVoid -> range.get());
+		return LanguageServers.forDocument(document).withCapability(ServerCapabilities::getLinkedEditingRangeProvider)
```

Comment:
The issue is that there is no filtering on the result: sometimes, some LS may return nothing or a single range and we need to use the next LS. This seems lost in translation. Maybe instead we should `computeAll` and filter the output to choose the most interesting one?

Reply:
Good catch, My code looked too simple :) I will fix that.